broadway/api/callbacks/job.py

```diff
@@ -56,6 +57,13 @@ def job_update_callback(settings, grading_job_id, grading_run_id):
             )
             return
 
+        stream_queue = settings["STREAM_QUEUE"]
+        if job.success:
+            stream_queue.update_job_state(job.id, GradingRunState.FINISHED)
+        else:
+            stream_queue.update_job_state(job.id, GradingRunState.FAILED)
+        stream_queue.send_close_event(job.id)
```

Comment:
I think we should move this chunk of code to before the `if` statement (line 32). This endpoint was not limited to student grading jobs by design. So if someone wants to use it for preprocessing jobs, it should also work. However, if we put this code here, then only student grading jobs will get updates. So we should move it up such that all jobs will get update.